source/common/network/connection_impl.cc

```diff
@@ -256,14 +269,43 @@ void ConnectionImpl::write(Buffer::Instance& data) {
     // ever changed, read the comment in Ssl::ConnectionImpl::doWriteToSocket() VERY carefully.
     // That code assumes that we never change existing write_buffer_ chain elements between calls
     // to SSL_write(). That code will have to change if we ever copy here.
-    write_buffer_->move(data);
+    write_buffer_.move(data);
 
     if (!(state_ & InternalState::Connecting)) {
       file_event_->activate(Event::FileReadyType::Write);
     }
   }
 }
 
+void ConnectionImpl::setBufferLimits(uint32_t limit) {
+  read_buffer_limit_ = limit;
+
+  // Due to the fact that writes to the connection and flushing data from the connection are done
+  // asynchronously, we have the option of either setting the watermarks aggressively, and regularly
+  // enabling/disabling reads from the socket, or allowing more data, but then not triggering
+  // based on watermarks until 2x the data is buffered in the common case.  Given these are all soft
+  // limits we err on the side of buffering more and having better performace.
+  // If the connection class is changed to write to the buffer and flush to the socket in the same
+  // stack, the high watermark should be changed to the buffer limit without the + 1
```

Comment:
Question: Not immediately obvious what the +1 is about. Can you explain a bit more in comment?